src/iapetos/collector/ring.clj

```diff
@@ -129,16 +129,59 @@
     (f)))
 
 (defn- run-instrumented
-  [{:keys [handler exception-status] :as options} request]
-  (ex/with-exceptions (exception-counter-for options request)
-    (let [start-time (System/nanoTime)
-          response   (safe exception-status #(handler request))
-          delta      (- (System/nanoTime) start-time)]
-      (->> (ensure-response-map response exception-status)
-           (record-metrics! options delta request))
-      (if-not (exception? response)
-        response
-        (throw response)))))
+  ([{:keys [handler exception-status] :as options} request]
+   (ex/with-exceptions (exception-counter-for options request)
+     (let [start-time (System/nanoTime)
+           response   (safe exception-status #(handler request))
+           delta      (- (System/nanoTime) start-time)]
+       (->> (ensure-response-map response exception-status)
+            (record-metrics! options delta request))
+       (if-not (exception? response)
+         response
+         (throw response)))))
+
+  ([{:keys [handler exception-status] :as options} request respond raise]
+   (let [start-time (System/nanoTime)
+         ex-counter (exception-counter-for options request)
+         respond-fn #(let [delta (- (System/nanoTime) start-time)]
+                       (->> (ensure-response-map % exception-status)
+                            (record-metrics! options delta request))
+                       (respond %))
+         raise-fn   #(do (ex/record-exception! ex-counter %)
```

Comment:
I would generally prefer `(fn [response] ...)` and `(fn [e] ...)` to `#(...)`. The `do` in `raise-fn` would be unnecessary then as well.

Reply:
I'll make that change, but I will point out that I didn't write this code.